Source/projections/Builder/ProjectionBuilder.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { Logger } from 'winston';
+
+import { Guid } from '@dolittle/rudiments';
+import { IEventTypes } from '@dolittle/sdk.artifacts';
+import { IContainer } from '@dolittle/sdk.common';
+import { ScopeId } from '@dolittle/sdk.events';
+import { ExecutionContext } from '@dolittle/sdk.execution';
+import { Cancellation } from '@dolittle/sdk.resilience';
+import { Constructor } from '@dolittle/types';
+
+import { ProjectionsClient } from '@dolittle/runtime.contracts/Runtime/Events.Processing/Projections_grpc_pb';
+
+import { IProjections, ProjectionId } from '../';
+
+import { ICanBuildAndRegisterAProjection } from './ICanBuildAndRegisterAProjection';
+import { ProjectionBuilderForReadModel } from './ProjectionBuilderForReadModel';
+import { ReadModelAlreadyDefinedForProjection } from './ReadModelAlreadyDefinedForProjection';
+
+export class ProjectionBuilder implements ICanBuildAndRegisterAProjection {
+    private _scopeId: ScopeId = ScopeId.default;
+    private _readModelTypeOrInstance?: Constructor<any> | any;
+    private _builder?: ProjectionBuilderForReadModel<any>;
+
+    /**
+     * Initializes a new instance of {@link ProjectionBuilder}.
+     * @param {ProjectionId} _projectionId  The unique identifier of the projection to build for
+     */
+    constructor(private _projectionId: ProjectionId) { }
+
+    /**
+     * Defines the projection to operate on a specific {@link ScopeId}.
+     * @param {ScopeId | Guid | string} scopeId Scope the projection operates on.
+     * @returns {ProjectionBuilder}
+     */
+    inScope(scopeId: ScopeId | Guid | string): ProjectionBuilder {
+        this._scopeId = ScopeId.from(scopeId);
+        return this;
+    }
+
+    /**
+     * Defines which readmodel to build a projection for. The initial readmodel will be generated with default values
+     * defined in the type.
```

Comment:
Something like:
Defines the type of the read model the projection builds. The initial state of a newly created read model is given by the provided instance or an instance constructed by the default constructor of the provided type.